zerver/lib/user_groups.py

```diff
@@ -464,12 +464,12 @@ def get_recursive_membership_groups(user_profile: UserProfile) -> QuerySet[UserG
 
 
 def is_user_in_group(
-    user_group: UserGroup, user: UserProfile, *, direct_member_only: bool = False
+    user_group: UserGroup, user_id: int, *, direct_member_only: bool = False
 ) -> bool:
     if direct_member_only:
-        return get_user_group_direct_members(user_group=user_group).filter(id=user.id).exists()
+        return get_user_group_direct_members(user_group=user_group).filter(id=user_id).exists()
 
-    return get_recursive_group_members(user_group=user_group).filter(id=user.id).exists()
+    return get_recursive_group_members(user_group=user_group).filter(id=user_id).exists()
```

Comment:
The changes  were made to maintain the optimization outlined in function `get_recipient_info` in `message_send.py`, as shown below:

```
query: ValuesQuerySet[UserProfile, ActiveUserDict] = UserProfile.objects.filter(
            is_active=True
        ).values(
            "id",
            "enable_online_push_notifications",
            "enable_offline_email_notifications",
            "enable_offline_push_notifications",
            "is_bot",
            "bot_type",
            "long_term_idle",
        )
```
As later in the function `get_recipient_info` we need to call `check_access_based_on_stream_topic_access_group` which would then call `is_user_in_group`.